http.c

```diff
@@ -1028,13 +1029,24 @@ evhttp_handle_chunked_read(struct evhttp_request *req, struct evbuffer *buf)
 			char *p = evbuffer_readln(buf, NULL, EVBUFFER_EOL_CRLF);
 			char *endp;
 			int error;
+			size_t len_p;
 			if (p == NULL)
 				break;
+			len_p = strlen(p);
 			/* the last chunk is on a new line? */
-			if (strlen(p) == 0) {
+			if (len_p == 0) {
 				mm_free(p);
 				continue;
 			}
+			error = isspace(p[0]) ||
```

Comment:
Let's add a comment for this hunk of code, also it will be great if you will add regression test. (`regress_http.c`)

Reply:
I have spent about an hour puzzling over `regress_http.c`, and I'm having trouble understanding how to add the test. I'm getting lost in a sea of callbacks.

Some guidance on an example test to base mine on would be helpful :)

Thanks

Reply:
https://github.com/libevent/libevent/pull/1552